### ISSUE TITLE
Synchronise trace timestamps with internal and other CDP timings

### ISSUE THREAD

**Issue by jahtalab** [elastic/synthetics#85]
Currently the Tracing event timestamps are not aligned with other events that are captured by the synthetic agent (e.g. step, performance, network). We need to fix this to be able to show the correct filmstrip for each step.

**Comment by vigneshshanmugam** [comment]
I am already investigating this issue #120 and probably will do a PR soonish, so will fix this issue now as its quite relevant for the other one and needs to be done before. Hope you are okay with it @jahtalab

**Comment by vigneshshanmugam** [comment]
Detailed analysis of how timestamps are created for trace events. 

* Chrome creates the trace events timestamp `ts` field based on the Internal `BaseClock` which is a monotonically non-decreasing clock time and platform dependant as explained in detail here -  https://source.chromium.org/chromium/chromium/src/+/master:base/time/time.h;l=936;bpv=0;bpt=0

If you want to dig more, go to the respective `time_[PLATFORM].cc` to check the implementation of each platform specific time ticks. 

* Another interesting fact is that, chrome tracing also supports synchronising clocks and fires [clock sync events](https://chromedevtools.github.io/devtools-protocol/tot/Tracing/#method-recordClockSyncMarker), So even if we spawn the chrome as a browser service, we can use this to merge the trace events and synchronise them with the other events respectively. 
I don't want to go in to more details, but if anyone is curious check the detailed document about [Tracing clock sync architecture](https://bit.ly/tracing-clock-sync).

* Chrome trace events are eventually moving towards a new model `Typed trace events` which is based on the [Perfecto tracing architecture](https://docs.google.com/document/d/1UQ4Ez7B-TeowijOUuMXuoWj1amZcQ7E2abt3s4jaAEY/edit?ts=5e81fcb8#), This is a long term plan and it would make consuming events much nicer. 

Chrometto(Chrome + Perfecto) effort is tracked on this document - https://docs.google.com/document/d/1f7tt4cb-JcA5bQFR1oXk60ncJPpkL02_Hi_Bc6MfTQk/

* Our internal based clocks are based on Node.js `hrtime` which is again a non decreasing monotonic clock which is platform dependent. 

This is only for our understanding and our current solution in #157 works for linux and mac which are our supported platforms at the moment. Let me know if anyone is interested in specific details.

**Comment by vigneshshanmugam** [comment]
Closing this as we have a workaround for now for our current supported platforms. Lets reopen a new issue if timestamps mismatch.
